pandas/tests/strings/test_find_replace.py

```diff
@@ -311,100 +406,185 @@ def test_replace_callable_raises(any_string_dtype, repl):
+def test_replace_literal(regex, expected, any_string_dtype):
     # GH16808 literal replace (regex=False vs regex=True)
-    values = Series(["f.o", "foo", np.nan], dtype=any_string_dtype)
-    expected = Series(["bao", "bao", np.nan], dtype=any_string_dtype)
-    result = values.str.replace("f.", "ba", regex=True)
+    ser = Series(["f.o", "foo", np.nan], dtype=any_string_dtype)
+    expected = Series(expected, dtype=any_string_dtype)
+    result = ser.str.replace("f.", "ba", regex=regex)
     tm.assert_series_equal(result, expected)
 
-    expected = Series(["bao", "foo", np.nan], dtype=any_string_dtype)
-    result = values.str.replace("f.", "ba", regex=False)
-    tm.assert_series_equal(result, expected)
 
-    # Cannot do a literal replace if given a callable repl or compiled
-    # pattern
-    callable_repl = lambda m: m.group(0).swapcase()
-    compiled_pat = re.compile("[a-z][A-Z]{2}")
+def test_replace_literal_callable_raises(any_string_dtype):
+    ser = Series([], dtype=any_string_dtype)
+    repl = lambda m: m.group(0).swapcase()
 
     msg = "Cannot use a callable replacement when regex=False"
     with pytest.raises(ValueError, match=msg):
-        values.str.replace("abc", callable_repl, regex=False)
+        ser.str.replace("abc", repl, regex=False)
+
+
+def test_replace_literal_compiled_raises(any_string_dtype):
+    ser = Series([], dtype=any_string_dtype)
+    pat = re.compile("[a-z][A-Z]{2}")
 
     msg = "Cannot use a compiled regex as replacement pattern with regex=False"
     with pytest.raises(ValueError, match=msg):
-        values.str.replace(compiled_pat, "", regex=False)
+        ser.str.replace(pat, "", regex=False)
+
+
+def test_replace_moar(any_string_dtype):
+    # PR #1179
+    ser = Series(
+        ["A", "B", "C", "Aaba", "Baca", "", np.nan, "CABA", "dog", "cat"],
+        dtype=any_string_dtype,
+    )
+
+    result = ser.str.replace("A", "YYY")
+    expected = Series(
+        ["YYY", "B", "C", "YYYaba", "Baca", "", np.nan, "CYYYBYYY", "dog", "cat"],
+        dtype=any_string_dtype,
+    )
+    tm.assert_series_equal(result, expected)
+
+    result = ser.str.replace("A", "YYY", case=False)
+    expected = Series(
+        [
+            "YYY",
+            "B",
+            "C",
+            "YYYYYYbYYY",
+            "BYYYcYYY",
+            "",
+            np.nan,
+            "CYYYBYYY",
+            "dog",
+            "cYYYt",
+        ],
+        dtype=any_string_dtype,
+    )
+    tm.assert_series_equal(result, expected)
+
+    result = ser.str.replace("^.a|dog", "XX-XX ", case=False, regex=True)
+    expected = Series(
+        [
+            "A",
+            "B",
+            "C",
+            "XX-XX ba",
+            "XX-XX ca",
+            "",
+            np.nan,
+            "XX-XX BA",
+            "XX-XX ",
+            "XX-XX t",
+        ],
+        dtype=any_string_dtype,
+    )
+    tm.assert_series_equal(result, expected)
+
+
+def test_replace_regex_default_warning(any_string_dtype):
```

Comment:
this one from pandas/tests/series/methods/test_replace.py and split and tested result